src/util.test.ts

```diff
@@ -217,3 +218,40 @@ describe('util', () => {
     })
   })
 })
+
+describe('safeRetry', () => {
+  test('should wrap string errors in Error object', async () => {
+    const fn = () => {
+      throw 'string error'
+    }
+    const safeFn = safeRetry(fn)
+    await assert.rejects(safeFn, { name: 'Error', message: 'string error' })
+  })
+
+  test('should pass through Error objects unchanged', async () => {
+    const originalError = new Error('test error')
+    const fn = () => {
+      throw originalError
+    }
+    const safeFn = safeRetry(fn)
+    await assert.rejects(safeFn, (err: Error) => {
+      assert.strictEqual(err, originalError)
+      return true
+    })
+  })
+
+  test('should wrap unknown errors in Error object', async () => {
+    const fn = () => {
+      throw { foo: 'bar' }
+    }
+    const safeFn = safeRetry(fn)
+    await assert.rejects(safeFn, { name: 'Error', message: 'Unknown error' })
```

Comment:
This behavior isn't totally satisfying because it obscures the original error. I wonder if it would be worthwhile to stringify objects, truncating them if they're really big. Or truncation could be left as the responsibility of the caller, like we tend to use `serializeError` in the Spark code base.

Reply:
The "cause" thing that @dshishkov mentioned might also solve this problem.